examples/mnist/tf_example.py

```diff
@@ -0,0 +1,125 @@
+#  Copyright (c) 2017-2018 Uber Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+###
+# Adapted to petastorm dataset using original contents from
+# https://github.com/tensorflow/tensorflow/blob/master/tensorflow/examples/tutorials/mnist/mnist_softmax.py
+###
+
+from __future__ import division, print_function
+
+import argparse
+import tensorflow as tf
+
+from examples.mnist import DEFAULT_MNIST_DATA_PATH
+from petastorm.reader import Reader
+
+
+def train_and_test(train_reader, test_reader, epochs, batch_size, log_interval):
+    """
+    Train a model for epochs with a batch size batch_size, printing accuracy every log_interval.
+    :param train_reader: A reader for train data.
+    :param test_reader: A reader for test data.
+    :param epochs: The epochs to train for.
+    :param batch_size: The batch size for training.
+    :param log_interval: The interval used to print the accuracy.
+    :return:
+    """
+    x = tf.placeholder(tf.float32, [None, 784])
+    W = tf.Variable(tf.zeros([784, 10]))
+    b = tf.Variable(tf.zeros([10]))
+    y = tf.matmul(x, W) + b
+
+    # Define loss and optimizer
+    y_ = tf.placeholder(tf.int64, [None])
+
+    # The raw formulation of cross-entropy,
+    #
+    #   tf.reduce_mean(-tf.reduce_sum(y_ * tf.log(tf.nn.softmax(y)),
+    #                                 reduction_indices=[1]))
+    #
+    # can be numerically unstable.
+    #
+    # So here we use tf.losses.sparse_softmax_cross_entropy on the raw
+    # outputs of 'y', and then average across the batch.
+    cross_entropy = tf.losses.sparse_softmax_cross_entropy(labels=y_, logits=y)
+    train_step = tf.train.GradientDescentOptimizer(0.5).minimize(cross_entropy)
+
+    sess = tf.InteractiveSession()
+    tf.global_variables_initializer().run()
+
+    # Train
+    print("Training model for {0} epochs with batch size {1} and log interval {2}".format(
+        epochs, batch_size, log_interval
+    ))
+    for i in range(epochs):
+        batch_x, batch_y = get_batch(train_reader, batch_size)
```

Comment:
Suggest we show our TF native interface:`tf_tensors` together with a standard `tf.train.batch`for batching. Here you use pure python API which is less interesting in our context (we don't want to show how to train MNIST, but to show how we hookup reader efficiently to train MNIST).